src/gedi_subset/subset.py

```diff
@@ -155,8 +157,7 @@ def subset_granule(props: SubsetGranuleProps) -> IOResultE[Maybe[str]]:
     GeoParquet file.
     """
 
-    io_result = download_granule(props.maap, str(props.output_dir), props.granule)
-    inpath = unsafe_perform_io(io_result.alt(raise_exception).unwrap())
+    inpath = download_granule(props.maap, str(props.output_dir), props.granule)
```

Comment:
Which of the following happens if we exhausted all the retries for a given granule : 

- The whole job fails
- We proceed with the other granules and the geopackage will be incomplete

The answer to this question wasn't easy to get by skimming (not very thoroughly) the code

Reply:
At the moment, if a single download fails, the entire job fails. To address this specific point, I created #62.

Given that, I chose to create a patient and persistent retry logic to greatly reduce the likelihood of download failures. While this may significantly increase the time a job runs, it should significantly increase the likelihood of job success. Given that this is a batch job system, and the overhead of submitting and checking jobs is non-trivial, users are generally happier to wait longer for a single job to succeed than to have to resubmit the job many times resulting in many failures before finally succeeding.

In the end, the longer wait for the first job to succeed is very likely still less time than submitting the job multiple times, particularly when it is common for users to post questions about job failures when they can't readily determine that a failure is transient, and that they simply need to keep retrying, which then takes unnecessary support time. Overall, this approach should be better for everybody (users _and_ user-support).

Having said all of that, we also have another ticket (can't find it offhand) that should completely circumvent this entire issue by using S3 rather than HTTPS for the collections that run into this problem.